modules/internal/String.chpl

```diff
@@ -835,7 +835,7 @@ module String {
         return '';
       } else if S.size == 1 {
         // TODO: ensures copy, clean up when no longer needed
-        var ret = S[S.domain.low];
```

Comment:
This was a bug before (and still is), but this should not work for the tuple-case... I guess we don't have a test for `string.join(tuple)` anywhere.